TransformationSystem/Client/TaskManagerPlugin.py

```diff
@@ -123,40 +129,27 @@ def _ByJobType( self ):
     elif 'ALL' in excludedSites:
       destSites = set()
     else:
-      destSites = destSites.difference( set( excludedSites ) )
+      destSites -= excludedSites
 
     # 4. get JobTypeMapping "Allow" section
     res = self.opsH.getOptionsDict( 'JobTypeMapping/%s/Allow' % jobType )
     if not res['OK']:
       gLogger.verbose( res['Message'] )
       allowed = {}
     else:
-      allowed = res['Value']
-      for site in allowed:
-        allowed[site] = fromChar( allowed[site] )
+      allowed = dict( ( site, set( fromChar( fromSites ) ) ) for site, fromSites in res['Value'].iteritems() )
 
     # 5. add autoAddedSites, if requested
-    autoAddedSites = self.opsH.getValue( 'JobTypeMapping/AutoAddedSites', [] )
```

Comment:
Removing this line is a bug, it seems to me (`autoAddedSites` won't be defined anymore)